libsolidity/formal/SymbolicState.h

```diff
@@ -27,55 +28,146 @@ namespace solidity::frontend::smt
 {
 
 class EncodingContext;
+class SymbolicAddressVariable;
+class SymbolicArrayVariable;
+
+class BlockchainVariable
+{
+public:
+	BlockchainVariable(std::string _name, std::map<std::string, smtutil::SortPointer> _members, EncodingContext& _context);
+	/// @returns the variable data as a tuple.
+	smtutil::Expression value() { return m_tuple->currentValue(); }
+	smtutil::Expression value(unsigned _idx) { return m_tuple->valueAtIndex(_idx); }
+	smtutil::SortPointer sort() { return m_tuple->sort(); }
+	unsigned index() { return m_tuple->index(); }
```

Comment:
```suggestion
	smtutil::Expression value() const { return m_tuple->currentValue(); }
	smtutil::Expression value(unsigned _idx) const { return m_tuple->valueAtIndex(_idx); }
	smtutil::SortPointer const& sort() const { return m_tuple->sort(); }
	unsigned index() const { return m_tuple->index(); }
```

Same with all other functions in `SymbolicState` that return the results from `sort()`, `currentValue()` and `valueAtIndex()`. For example `sort()` is defined in `SymbolicVariable` as:
```c++
	smtutil::SortPointer const& sort() const { return m_sort; }
```

Reply:
You missed the `const&` after `smtutil::SortPointer`. It's not a big difference and perhaps just a microptimization but `m_tuple->sort()` is defined that way so there's not reason for this one to be different.

Reply:
Right!